alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionManager.java

```diff
@@ -0,0 +1,554 @@
+            }
+        }
+
+        private void evaluateEach(String target, Map<String, Map<String, List<? extends BucketPoint>>> queryResults,
+                ConditionEvaluator evaluator) {
+
+            log.infof("");
+            // get the Set of metrics common to all of the queries
+            Set<String> metrics = null;
+            for (String queryName : queryResults.keySet()) {
+                if (null == metrics) {
+                    metrics = new HashSet<>(queryResults.get(queryName).keySet());
+                    continue;
+                }
+                metrics.retainAll(queryResults.get(queryName).keySet());
+            }
+
+            // evaluate each
+            Map<String, BucketPoint> metricQueryResults = new HashMap<>();
+            Map<String, String> preparedCondition = new HashMap<>();
+            for (String metric : metrics) {
+                metricQueryResults.clear();
+                preparedCondition.clear();
+
+                for (String queryName : queryResults.keySet()) {
+                    metricQueryResults.put(queryName, queryResults.get(queryName).get(metric).get(0));
+                }
+
+                try {
+                    preparedCondition = evaluator.prepare(metricQueryResults);
+                } catch (Exception e) {
+                    log.debugf("Could not prepare evaluation, Skipping due to [%s]", e.getMessage());
+                }
+
+                if (evaluator.evaluate()) {
+                    try {
+                        Event externalEvent = new Event(externalCondition.getTenantId(), UUID.randomUUID().toString(),
+                                System.currentTimeMillis(), externalCondition.getDataId(),
+                                TAG_EXTERNAL_CONDITION_VALUE,
+                                preparedCondition.toString(),
+                                Collections.singletonMap("metricName", metric), null);
+                        if (log.isDebugEnabled()) {
+                            log.debug("Sending External Condition Event to Alerting! " + externalEvent);
+                        }
+                        alertsService.sendEvents(Collections.singleton(externalEvent));
+
+                    } catch (Exception e) {
+                        log.error("Failed to send external event to alerting system.", e);
+                    }
+                }
+            }
+        }
+
+        private void evaluate(String target, Map<String, BucketPoint> queryResults, ConditionEvaluator evaluator) {
+            Map<String, String> preparedCondition = null;
+            try {
+                preparedCondition = evaluator.prepare(queryResults);
+            } catch (Exception e) {
+                log.debugf("Could not prepare evaluation, Skipping due to [%s]", e.getMessage());
+            }
+
+            if (evaluator.evaluate()) {
+                try {
+                    Data externalData = Data.forString(externalCondition.getTenantId(),
```

Comment:
So, I am not 100% sure if I understand well some details of the logic.
But in EACH mode, the ExternalCondition will expect and Event and in ALL mode a Data to trigger the Alert.
Am I right ?